Orange/evaluation/testing.py

```diff
@@ -289,59 +348,26 @@ def __init__(self, data, learners, k=10, stratified=True, random_state=0, store_
         super().__init__(data, len(learners), store_data=store_data,
                          store_models=store_models, preprocessor=preprocessor,
                          callback=callback)
+        self.folds = []
+        self.learners = learners
         self.k = k
         self.stratified = stratified
         self.random_state = random_state
 
-        indices = None
+        self.indices = None
         if self.stratified and data.domain.has_discrete_class:
-            indices = skl_cross_validation.StratifiedKFold(
+            self.indices = skl_cross_validation.StratifiedKFold(
                 data.Y, self.k, shuffle=True, random_state=self.random_state
             )
-            if any(len(train) == 0 or len(test) == 0 for train, test in indices):
+            if any(len(train) == 0 or len(test) == 0 for train, test in self.indices):
                 if warnings is not None:
                     warnings.append("Using non-stratified sampling.")
-                indices = None
```

Comment:
If any of the folds is empty, use non-stratified sampling. This is what the warning says. `self.indices` must be set to `None` for the behavior to match.